### PR TITLE
Add a big note with example about deprecated share() method

### PR DESCRIPTION
**Pull request by bocharsky-bw**
In complement to the note in https://github.com/knpuniversity/dependency-injection/pull/10

### CODE REVIEW

**Comment by bocharsky-bw** [comment]
Just checked - it looks like:

screen shot 2017-01-19 at 22 27 40

File: challenges.yml [hunk 1]
```diff
@@ -259,9 +259,24 @@ injectioninterfaces_interfacetypehintcoding:
 # Container
 container_moveservicecreationtocontainercoding:
     question: |
-        Ok, it's time to get organized! You've already installed the
-        `pimple/pimple` library *and* created your `$container` variable.
+        Ok, it's time to get organized! You've already installed the `pimple/pimple`
+        library which has version 3.0 *and* created your `$container` variable.
```
Comment (weaverryan) [inline challenges.yml]: ... library at version 3.0 *and* ...

File: challenges.yml [hunk 2]
```diff
@@ -259,9 +259,24 @@ injectioninterfaces_interfacetypehintcoding:
...
+        ```php
+        $container = new Container();
+        $container['mailer'] = function() {
+            return new Mailer();
+        };
+        ```
+
+        The `mailer` service will be created only once on the first call no matter
+        how many times you'll call it.
+        ***
```
Comment (weaverryan) [inline challenges.yml:279]: > In Pimple 3.0, you don't need to wrap your anonymous function with the `share()` method anymore. Now, it's much simpler!

(then after the code).

> Like in earlier Pimple versions, the `mailer` service will still be created only once...

**Review by weaverryan — COMMENTED** [review]
It looks great! 👍 for this PR and the other one! Thank you!

**Pull request merged**